ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/block/ReexecutingExecutionPayloadBlockManager.java

```diff
@@ -101,16 +116,32 @@ public SafeFuture<BlockImportResult> importBlock(SignedBeaconBlock block) {
         .thenPeek(
             blockImportResult -> {
               if (requiresReexecution(blockImportResult)) {
-                pendingBlocksForEPReexecution.add(block);
+                pendingBlocksForEPReexecution.put(block, block.getSlot().plus(RETRY_SLOTS));
               }
             });
   }
 
+  public void requireAncestor(final Bytes32 root) {
+    if (pendingBlocksForEPReexecution.isEmpty()) {
+      return;
+    }
+    final Bytes32 parentForFirstPending = pendingBlocks.getFirstRequiredAncestor(root);
+    final Optional<SignedBeaconBlock> reExecuteBlock =
+        pendingBlocksForEPReexecution.keySet().stream()
+            .filter(
+                block -> BLOCK_HASH_TREE_ROOT_FUNCTION.apply(block).equals(parentForFirstPending))
+            .findFirst();
+    reExecuteBlock.ifPresent(
+        block -> pendingBlocksForEPReexecution.put(block, currentSlot.plus(RETRY_SLOTS)));
```

Comment:
This seems to assume that we already have the block and are just extending the expiry when we receive new blocks.  What happens if the block is dropped from `pendingBlocksForEPReexecution` before we receive one of its ancestors? It seems like we'd get stuck again.